packages/shared-core/src/filters.ts

```diff
@@ -160,6 +167,7 @@ type QueryFields = {
 
 type QueryConfig = {
   allOr?: boolean
+  onEmptyFilter?: EmptyFilterOption
```

Comment:
I don't think we need this in here - this type is mixed with `QueryFields` to represent the available options for `operator`, which is how we pass up the flag for whether we have an AND or and OR query, i.e. `operator = "allOr"` indicates that it's an OR.

For this new field, we set it top level in the filter object (hence it's presence in the `Filter` type), but we don't need to add it into the `QueryConfig` type since we will never send up `operator = "onEmptyFilter"`.

Reply:
As discussed this was needed in the end. I have pushed up further code for handling SQL tables, and in the case of custom queries the empty filter setting is hidden.